### DISCUSSION TITLE
How to get AuthUser in onRecordBeforeUpdateRequest?

### DISCUSSION THREAD

**Discussion by Sadragos** [pocketbase/pocketbase#3031] [Q&A] [answered]
First of: Thanks for the great Software and the even better update - simple hooks and triggers was the only thing I was missing from PB - with its addition this tool ist just perfect for me!

I want to extend the base Model of every collection in my app to inlcude createdBy and updatedBy. These fields should be automatically filled, just like the date-fields. To fill them the AuthUser should be used. Is that currently possible with the hooks? I was hoping to find some auth-data within the httpContext attribute, but that seems to always be empty.

**Comment by ganigeorgiev** [comment, marked as answer]
You can define the `createdBy` and the `updatedBy` as `relation` fields pointing to your auth collection and then you can attach to the [`onRecordBeforeCreateRequest`](https://pocketbase.io/docs/js-event-hooks/#onrecordbeforecreaterequest) and [`onRecordBeforeUpdateRequest`](https://pocketbase.io/docs/js-event-hooks/#onrecordbeforeupdaterequest) hooks. The request context in the hooks is available under `e.httpContext` (this is the echo/router instance, aka. the `c` variable in the routing examples - https://pocketbase.io/docs/js-routing/#retrieving-the-current-auth-state).

Here is a code sample:

```js
onRecordBeforeCreateRequest((e) => {
    const admin  = e.httpContext.get("admin")
    if (admin) {
        return; // admins are allowed to do anything
    }

    const authRecord = e.httpContext.get("authRecord")
    if (authRecord) {
        e.record.set("createdBy", authRecord.id)
        e.record.set("updatedBy", authRecord.id)
    } else {
        // maybe throw new ForbiddenError() or leave the api rule to handle this case?
    }
}, "collectionA", "collectionB")

onRecordBeforeUpdateRequest((e) => {
    const admin  = e.httpContext.get("admin")
    if (admin) {
        return; // admins are allowed to do anything
    }

    const authRecord = e.httpContext.get("authRecord")
    if (authRecord) {
        e.record.set("updatedBy", authRecord.id)
    } else {
        // maybe throw new ForbiddenError() or leave the api rule to handle this case?
    }
}, "collectionA", "collectionB")
```

**Reply by Sadragos** [reply]
Wow that was fast! Thanks a lot, that's exactly what I was looking for.
